### DISCUSSION TITLE
What is the blank area on the left and right of the DataGridCell that cannot be eliminated?

### DISCUSSION THREAD

**Discussion by abcbuzhiming** [AvaloniaUI/Avalonia#16379] [Q&A] [answered]
I ues Avalonia.Controls.DataGrid.  During use, I found the following phenomenon, as shown in the following figure。
![01](https://github.com/user-attachments/assets/1a3a6575-2e42-4f36-83dc-d9e772094790)
There is a blank area on the left and right sides of each cell. When the cell width is sufficient, this blank area has no effect.

But when you reduce the width of the cell, as shown in the figure below, this blank area will not only prevent the content from getting close to the edge of the table, but also block the content.
![02](https://github.com/user-attachments/assets/a8554afd-57e0-4d9c-8cf2-ab0107a557fd)

I tried many ways. I set the DataGridSortIconMinWidth value to 0, and I set the DataGridCell's Padding and Margin to 0.But it doesn't work。

So what is this and is there any way to eliminate it? because this white space takes up extra width and prevents me from having a compact table.

**Comment by rabbitism** [comment, marked as answer]
I think this is the line that controls this margin. You can try to redefine this theme in your resource dictionary. 
https://github.com/AvaloniaUI/Avalonia/blob/890df998a9273ec25e682585346b1e76678b4c1f/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L72-L75

**Reply by abcbuzhiming** [reply]
Add this part to Application.Resources and reset Margin to solve the problem.

Thanks you very much
